cast/src/lib.rs

```diff
@@ -447,8 +453,40 @@ impl SimpleCast {
         format!("0x{}", s)
     }
 
+    pub async fn generate_interface(
+        address_or_path: String,
+        chain: Chain,
+        etherscan_api_key: Option<String>,
+    ) -> Result<Vec<InterfaceSource>> {
+        let mut contract_names: Vec<String> = Vec::new();
+        let contract_abis: Vec<Abi> = if Path::new(&address_or_path).exists() {
+            println!("Reading ABI from local file at {}", &address_or_path);
+            contract_names.push("Interface".to_owned());
+            let file = std::fs::read_to_string(&address_or_path).expect("unable to read abi file");
```

Comment:
```suggestion
            let file = std::fs::read_to_string(&address_or_path)?;
```

you can also use `.wrap_err(...)?` to add extra context to your errors